model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/entities/LoginFailureEntity.java

```diff
@@ -88,7 +88,9 @@ public long getLastFailure() {
     }
 
     public void setLastFailure(long lastFailure) {
-        this.lastFailure = lastFailure;
+        if(lastFailure>this.lastFailure) {
+            this.lastFailure = lastFailure;
+        }
```

Comment:
@douglaspalmer Probably we need something similar in `failedLoginNotBefore`.

Reply:
Done